Reduce status text spam (e.g. load/save world)

SignatureBeef opened the pull request:
First attempt at implementing #2560

Additional changes: Potentially improve debug builds not copying the plugin to ServerPlugins

![image](https://user-images.githubusercontent.com/776327/156110702-0915defb-43f6-4d81-b5c4-5b0e5f608039.png)


<!-- Warning: If you create a pull request and wish to remain anonymous, you are highly advised to use Tails (https://tails.boum.org/) or a fresh git environment. We will *not* be able to help with anonymization after your pull request has been created. -->

<!-- Warning: If you do not update the changelog, your pull request will be ignored and eventually closed, without comment, without any support, and without any opinion or interaction from the TShock team. This is your only warning. -->

hakusaro commented:
@SignatureBeef protip if you put "fixes #2560" in your commit it'll close the issue when the branch is merged

hakusaro commented on TShockAPI/TShock.cs:
```diff
@@ -442,6 +444,41 @@ public override void Initialize()
 			}
 		}
 
+		/// <summary>
```
Is there any way we can move this method + the related variables/hooks into any other file that isn't `TShock.cs`? It's a minor nit but the file is giant and I think it would be better served as a "micro-plugin" that lives in its own file somewhere, even if we have to spend some work initializing it here.

SignatureBeef replied:
i've implemented a module system via d2a060b, and then the new functionality in a single file via fb67fcc
Hopefully this is what you are meaning by micro plugin - these modules can be a single file and will be discovered automatically. Im thinking over time TShock.cs can be split up to use this same system.

hakusaro commented on TShockLauncher/TShockLauncher.csproj:
```diff
@@ -32,8 +32,10 @@
 	<Target Name="MoveTShockDebug" AfterTargets="FinalCleanup;PostBuildEvent">
 		<ItemGroup>
 			<TShockPluginFilesDebug Include="$(TargetDir)/TShockAPI*" />
+			<TShockPluginFilesDebugNonPublish Include="$(ProjectDir)../TShockAPI/bin/$(Configuration)/$(TargetFramework)/TShockAPI*" Condition="'$(PublishSingleFile)' != 'true'" />
```
Can this change be broke into a separate commit? I definitely initially thought this was a misplaced addition.

SignatureBeef replied:
I've dropped this until i can test it further, and it's not specific for this PR.

hakusaro commented on CHANGELOG.md:
```diff
@@ -12,6 +12,9 @@ This is the rolling changelog for TShock for Terraria. Use past tense when addin
   * If there is no section called "Upcoming changes" below this line, please add one with `## Upcoming changes` as the first line, and then a bulleted item directly after with the first change.
 
+## Upcoming changes (TShock 5.0.0)
+* Reduced load/save console spam
```
```suggestion
* Reduced load/save console spam. (@SignatureBeef, @YehnBeep)
```

I don't know that this is going to be clear enough for average people to understand, but with names + formatting it's okay.

hakusaro reviewed (commented):
Thank you so much for your work on this! I have minor nits/suggestions, but nothing major as a deal breaker. We can ultimately accept these changes easily, but I'd prefer a couple changes. If you don't want to make changes, lmk and I'll swap to an lgtm and merge + change stuff.

SignatureBeef committed:
Create module subsystem under TShockAPI.Modules

This should allow for TShock.cs to be split up to mini-modules, or "micro-plugins"

SignatureBeef committed:
Create module to reduce console spam - fixes #2560

SignatureBeef committed:
Update CHANGELOG.md

SignatureBeef committed:
Update CHANGELOG.md

Co-authored-by: Lucas Nicodemus <shank@shanked.me>

hakusaro reviewed (approved):
![lgtm_trick](https://user-images.githubusercontent.com/532647/157828044-11172f8f-d6d9-4742-b6b7-ecfc8ff4a735.gif)


Sorry, I was in San Francisco, but I'm back at my satellite office and I'm happy to say this is AWESOME and it sets a good precedent for the future. Cheers! I love you!
